### DISCUSSION TITLE
Projects in release mode cannot print to the terminal using print

### DISCUSSION THREAD

**Discussion by HalfSweet** [tauri-apps/tauri#8626] [Q&A] [answered]
I'm trying to build a mix of cli and gui, using cli for interaction when starting the project in the terminal and parameters are present, otherwise launching a tauri window.

Here is my code

```rust
fn main() {
    println!("Main start");
    let args: Vec<String> = std::env::args().collect();
    if args.len() > 1 {
        println!("Hello, {}", args[1]);
        eprintln!("Hello, {}", args[1]);
        return;
    }
    tauri::Builder::default()
        .invoke_handler(tauri::generate_handler![greet])
        .run(tauri::generate_context!())
        .expect("error while running tauri application");
}
```

Frustratingly, it runs fine in debug mode, but in release mode the terminal doesn't print any valid information

![图片](https://github.com/tauri-apps/tauri/assets/60973476/15d668eb-248e-4a34-92e7-d8f0f1825051)

**Comment by amrbashir** [comment, marked as answer]
See https://github.com/tauri-apps/tauri/issues/8305#issuecomment-1826871949
